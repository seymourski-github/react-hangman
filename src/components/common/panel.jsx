import React from 'react';
import PropTypes from 'prop-types';

const renderProps = (id, props) => {
  return props.render(id, {
    ...props,
    id: `${id}-content`
  });
};

export const Panel = props => {
  const { className, id, children, render } = props;

  if(!render && !children) return;

  return (
    <div className={className} id={id}>
      <div className={`${className}-content`}>
        {children || renderProps(id, props)}
      </div>
    </div>
  );
};

Panel.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string
  ]),
  id: PropTypes.string.isRequired,
  render: PropTypes.func,
};

Panel.defaultProps = {
  className: 'panel',
  render: () => null
};
