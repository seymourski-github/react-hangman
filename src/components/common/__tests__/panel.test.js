import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Panel } from '../panel';

const props = {
  id: 'my-panel' 
};

describe('<Panel /> Component', function() {
  const component = renderer.create(<Panel {...props} />);

  it("renders and matches our snapshot", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
