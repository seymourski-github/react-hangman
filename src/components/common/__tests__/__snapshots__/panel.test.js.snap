// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> Component renders and matches our snapshot 1`] = `
<div
  className="panel"
  id="my-panel"
>
  <div
    className="panel-content"
  />
</div>
`;
